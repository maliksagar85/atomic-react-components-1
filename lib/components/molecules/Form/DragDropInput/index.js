export { default } from './DragDropInput';
export { DDFieldInputVanilla } from './DragDropInput';
