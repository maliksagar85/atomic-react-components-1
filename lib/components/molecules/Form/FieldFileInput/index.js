export { default } from './FieldFileInput';
export { FieldFileInputVanilla } from './FieldFileInput';
