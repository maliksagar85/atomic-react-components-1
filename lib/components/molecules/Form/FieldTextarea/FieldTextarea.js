import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';
import Textarea from '../../../atoms/Textarea';
import FieldError from '../../../atoms/FieldError';
import Label from '../../../atoms/Label';
import styles from './FieldTextarea.style';

const FieldTextarea = ({
  id,
  className,
  name,
  onChange,
  label,
  labelProps,
  register,
  errors,
  ...others
}) => (
  <div className={className}>
    <Label htmlFor={id} {...labelProps}>
      {label}
    </Label>
    <Textarea name={name} ref={register} {...others} />
    {errors[name] && <FieldError>{errors[name].message}</FieldError>}
  </div>
);

FieldTextarea.propTypes = {
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  label: PropTypes.node.isRequired,
  labelProps: PropTypes.objectOf(PropTypes.object),
  value: PropTypes.string,
  onChange: PropTypes.string,
  register: PropTypes.func.isRequired,
  errors: PropTypes.objectOf(PropTypes.object),
};

FieldTextarea.defaultProps = {
  className: '',
  value: '',
  onChange: null,
  labelProps: {},
  errors: {},
};

const StyledFieldTextarea = styled(FieldTextarea)`
  ${styles};
`;

export default StyledFieldTextarea;

export { FieldTextarea as FieldTextareaVanilla };
