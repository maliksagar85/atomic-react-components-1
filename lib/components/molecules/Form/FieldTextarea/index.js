export { default, FieldTextareaVanilla } from './FieldTextarea';
