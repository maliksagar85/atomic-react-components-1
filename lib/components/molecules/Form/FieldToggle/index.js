export { default } from './FieldToggle';
export { FieldToggleVanilla } from './FieldToggle';
