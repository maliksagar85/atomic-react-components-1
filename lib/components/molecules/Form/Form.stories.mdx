import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import {
  userFormProps,
  userFormSelectProps,
  fileMultipleInputProps,
  fileSingleInputProps,
  fileDDInputProps,
  userFormNameInputProps,
  userFormEmailProps,
  userFormCommentProps,
  userFormAgeGroupProps,
  userFormTnCProps,
  userFormToggleProps,
} from './Form.mock';
import Form from './Form'
import Button from '../../atoms/Button';

<Meta title="Molecules/Form" component={Form} />

### Default

<Preview>
  <Story name="Default">
    <Form {...userFormProps}>
      <Form.Select {...userFormSelectProps} />
      <Form.Input {...userFormNameInputProps} />
      <Form.Input {...userFormEmailProps} />
      <Form.RadioGroup {...userFormAgeGroupProps} />
      <Form.Checkbox {...userFormTnCProps} />
      <Form.Textarea {...userFormCommentProps} />
      <Form.Toggle {...userFormToggleProps} />
      <Form.FieldFile {...fileSingleInputProps} />
      <Form.FieldFile {...fileMultipleInputProps} />
      <Form.DDFile {...fileDDInputProps} />
      <Button primary type="submit">
        Submit
      </Button>
    </Form>
  </Story>
</Preview>

## Props ðŸ”§

<Props of={Form} />


