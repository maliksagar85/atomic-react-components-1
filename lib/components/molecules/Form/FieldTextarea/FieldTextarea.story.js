// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import FieldTextarea from '.';
import Form from '../Form';
import defaultTextareaForm from './FieldTextarea.mock';

storiesOf('Molecules/FieldTextarea', module)
  .addParameters({ jest: ['FieldTextarea', 'Form'] })
  .add('Default', () => (
    <Form {...defaultTextareaForm}>
      <FieldTextarea
        id="comment"
        name="comment"
        label="Comment"
        placeholder="Enter your text here..."
      />
    </Form>
  ));
