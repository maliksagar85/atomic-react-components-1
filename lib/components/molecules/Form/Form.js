import React from 'react';
import { useForm } from 'react-hook-form';
import PropTypes from 'prop-types';

import { buildSchema } from '../../../core/utils/buildValidationSchema';
import Debug from './Debug';
import FieldInput from './FieldInput';
import FieldSelect from './FieldSelect';
import FieldTextarea from './FieldTextarea';
import FieldRadioGroup from './FieldRadioGroup';
import FieldCheckbox from './FieldCheckbox';
import FieldToggle from './FieldToggle';
import FieldFileInput from './FieldFileInput';
import DDFileInput from './DragDropInput';

const Form = ({
  onSubmit,
  initialValues,
  children,
  debug,
  validate,
  validationSchema,
  yupValidationSchema,
}) => {
  const getSchema = () => {
    if (yupValidationSchema) return yupValidationSchema;
    return validationSchema ? buildSchema(validationSchema) : null;
  };
  const { handleSubmit, register, errors, setValue } = useForm({
    mode: validate,
    validationSchema: getSchema(yupValidationSchema, validationSchema),
    defaultValues: initialValues,
  });

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      {React.Children.map(children, (child) =>
        React.cloneElement(child, { register, errors, setValue })
      )}
      {debug && <Debug />}
    </form>
  );
};

Form.Checkbox = FieldCheckbox;

Form.Input = FieldInput;

Form.RadioGroup = FieldRadioGroup;

Form.Select = FieldSelect;

Form.Textarea = FieldTextarea;

Form.Toggle = FieldToggle;

Form.FieldFile = FieldFileInput;

Form.DDFile = DDFileInput;

Form.defaultProps = {
  debug: false,
  initialValues: null,
  validationSchema: null,
  yupValidationSchema: null,
  validate: 'onSubmit',
};

Form.propTypes = {
  initialValues: PropTypes.objectOf(PropTypes.object),
  children: PropTypes.node.isRequired,
  onSubmit: PropTypes.func.isRequired,
  debug: PropTypes.bool,
  validationSchema: PropTypes.objectOf(PropTypes.object),
  validate: PropTypes.string,
  yupValidationSchema: PropTypes.func,
};

export default Form;
