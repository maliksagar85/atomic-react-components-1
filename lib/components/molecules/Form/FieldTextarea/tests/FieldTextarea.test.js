import React from 'react';
import { shallowWithTheme, renderWithTheme } from '../../../../../../__mock__/themeMock';
import { FieldTextareaVanilla } from '../FieldTextarea';
import Form from '../../Form';
import { ButtonVanilla } from '../../../../atoms/Button';

describe('FieldTextarea Component', () => {
  let FieldTextarea = '';

  test('should render correctly', () => {
    FieldTextarea = shallowWithTheme(
      <Form>
        <div>
          <FieldTextareaVanilla />
          <FieldTextareaVanilla />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    expect(FieldTextarea).toMatchSnapshot();
  });

  test('should render label correctly', () => {
    const props = {
      label: 'Comment',
      name: 'comment',
    };
    FieldTextarea = renderWithTheme(
      <Form>
        <div>
          <FieldTextareaVanilla {...props} />
          <FieldTextareaVanilla {...props} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    expect(FieldTextarea.find('label').text()).toEqual('CommentComment');
  });

  test('should render textarea tag', () => {
    const props = {
      label: 'Comment',
      name: 'comment',
    };
    FieldTextarea = renderWithTheme(
      <Form>
        <div>
          <FieldTextareaVanilla {...props} />
          <FieldTextareaVanilla {...props} />
        </div>
        <ButtonVanilla primary type="submit">
          Submit
        </ButtonVanilla>
      </Form>
    );
    expect(FieldTextarea.find('textarea')).toHaveLength(2);
  });
});
