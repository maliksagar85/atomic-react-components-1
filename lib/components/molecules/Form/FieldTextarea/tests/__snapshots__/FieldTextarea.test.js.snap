// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldTextarea Component should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    errors={Object {}}
    key=".0"
    register={[Function]}
    setValue={[Function]}
  >
    <FieldTextarea
      className=""
      errors={Object {}}
      labelProps={Object {}}
      onChange={null}
      value=""
    />
    <FieldTextarea
      className=""
      errors={Object {}}
      labelProps={Object {}}
      onChange={null}
      value=""
    />
  </div>
  <Button
    ariaLabel=""
    className=""
    disabled={false}
    elementLocator=""
    errors={Object {}}
    inheritedStyles=""
    key=".1"
    primary={true}
    register={[Function]}
    secondary={false}
    setValue={[Function]}
    tertiary={false}
    transparent={false}
    type="submit"
    variation=""
  >
    Submit
  </Button>
</form>
`;
